geometry/line_segment_intersection.cpp

```diff
@@ -0,0 +1,88 @@
+#include <iostream>
+
+struct Point {
```

Comment:
struct is undocumented.
you can add a brief comment like so: [`others/smallest_circle`](https://github.com/kvedala/C-Plus-Plus/blob/master/others/smallest_circle.cpp)